api/src/transactions.rs

```diff
@@ -1182,14 +1182,19 @@ impl TransactionsApi {
         // Simulate transaction
         let state_view = self.context.latest_state_view_poem(&ledger_info)?;
         let move_resolver = state_view.as_move_resolver();
-        let (_, output_ext) = AptosVM::simulate_signed_transaction(&txn, &move_resolver);
+        let (_, vm_output) = AptosVM::simulate_signed_transaction(&txn, &move_resolver);
         let version = ledger_info.version();
 
-        // Apply transaction outputs to build up a transaction
-        // TODO: while `into_transaction_output_with_status()` should never fail
-        // to apply deltas, we should propagate errors properly. Fix this when
-        // VM error handling is fixed.
-        let output = output_ext.into_transaction_output(&move_resolver);
+        let output = match vm_output.try_materialize(&move_resolver) {
```

Comment:
Why not do this whole block inside simulate_signed_transaction? esp. since the error mentions during transaction simulation.